### PR TITLE
feat: note on DA, DAS, erasure codes

### PR DESCRIPTION
**Pull request by ultraviolet10**
Adds compiled notes on data availability, sampling, and the concept of erasure codes.

### CODE REVIEW

File: docs/wiki/research/scaling/data-availability.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+We require blocks to commit to the Merkle root of this "extended" data, and have light clients probabilistically check that the majority of the extended data is available, on which one of the three things is true
+
+- The entire extended data is available, the erasure code is constructed correctly, and the block is valid.
+- The entire extended data is available, the erasure code is constructed correctly, but the block is invalid.
+- The entire extended data is available, but the erasure code is constructed incorrectly.
+
+In case (1), the block is valid and the light client can accept it. In case (2), it is expected that some other node will quickly construct and relay a fraud proof. In case (3), it is also expected that some other node will quickly construct and relay a specialized kind of fraud proof that shows that the erasure code is constructed incorrectly. If a light client receives no fraud proofs for some time, it will take that as evidence that the block is in fact valid.
+
+An interesting implementation to study can be found [here](https://github.com/ethereum/research/tree/master/erasure_code/ec65536).
```
Comment (taxmeifyoucan) [inline docs/wiki/research/scaling/data-availability.md:49]: Please add a section for resources. There are many great articles on DA and solutions you explained here. Ethorg does a great job on DA https://ethereum.org/en/developers/docs/data-availability/

**Review by taxmeifyoucan — COMMENTED** [review]
Thanks for the contribution!

File: docs/wiki/research/scaling/data-availability.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Data Availability
+
+A core responsibility of any layer-1 blockchain is to guarantee _data availability_.
+
+Data availability refers to the availability of transactions in a block that is appended to the tip of the chain. During consensus, validators download the block to verify its availability. If the block contains any transactions that are withheld by a validator, the block is unavailable and will be rejected as invalid.
```
[the remark this answers is not in the document]
Comment (ultraviolet10) [reply docs/wiki/research/scaling/data-availability.md:5]: Do you suggest I collate the exact rules that are being verified?

File: docs/wiki/research/scaling/data-availability.md (hunk 1 above)
Comment (ultraviolet10) [reply docs/wiki/research/scaling/data-availability.md:49]: Added a few interesting ones, lmk what you think!

File: docs/wiki/research/scaling/data-availability.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (taxmeifyoucan) [reply docs/wiki/research/scaling/data-availability.md:5]: No need to go to details. Just to point out that a node needs to verify transactions and validate blocks to be able to operate trustlessly and build the state which requires historical and latest data.

**Comment by ultraviolet10** [comment]
took your suggestion and added it in verbatim 😅 @taxmeifyoucan

**Comment by ultraviolet10** [comment]
> Thanks @ultraviolet10! One paragraph I would find missing is the changes to L2 data availability with introduction of ephemeral blobs

Got it, I'll add this in the next update. Thanks :)

**Pull request merged**